packages/merge-on-green/src/merge-logic.ts

```diff
@@ -0,0 +1,537 @@
+/**
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GitHubAPI } from 'probot/lib/github';
+
+interface CheckParam {}
+
+interface CheckRun {
+  name: string;
+  conclusion: string;
+}
+
+interface CheckStatus {
+  context: string;
+  state: string;
+}
+interface Reviews {
+  user: {
+    login: string;
+  };
+  state: string;
+}
+
+interface ReviewsRequested {
+  users: [];
+  teams: [];
+}
+
+interface PR {
+  title: string;
+  body: string;
+}
+
+interface RequiredChecksByLanguage {
+  [key: string]: {
+    requiredStatusChecks: string[];
+  };
+}
+
+interface Language {
+  language: string;
+  repo: string;
+}
+
+// contains the installation id necessary to authenticate as an installation
+mergeOnGreen.getLatestCommit = async function getLatestCommit(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<string> {
+  try {
+    // TODO: consider switching this to an async iterator, which would work
+    // for more than 100.
+    const data = await github.pulls.listCommits({
+      owner,
+      repo,
+      pull_number: pr,
+      per_page: 100,
+      page: 1,
+    });
+    return data.data[data.data.length - 1].sha;
+  } catch (err) {
+    return '';
+  }
+};
+
+mergeOnGreen.getPR = async function getPR(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<PR> {
+  try {
+    const data = await github.pulls.get({
+      owner,
+      repo,
+      pull_number: pr,
+    });
+    return data.data;
+  } catch (err) {
+    return { title: '', body: '' };
+  }
+};
+
+mergeOnGreen.getMOGLabel = async function getMOGLabel(
```

Comment:
abbreviation in function name

Also, we should probably name this `hasMergeOnGreenLabel` as it returns a boolean.